Assets/BossRoom/Scripts/Server/Game/Character/ServerCharacter.cs

```diff
@@ -105,6 +106,31 @@ public override void OnNetworkDespawn()
             }
         }
 
+        void InitializeHitPoints()
+        {
+            NetState.HitPoints = NetState.CharacterClass.BaseHP.Value;
+
+            if (!IsNpc)
+            {
+                SessionPlayerData? sessionPlayerData = SessionManager<SessionPlayerData>.Instance.GetPlayerData(OwnerClientId);
+                if (sessionPlayerData is {HasCharacterSpawned: true})
+                {
+                    NetState.HitPoints = sessionPlayerData.Value.CurrentHitPoints;
+                    if (NetState.HitPoints <= 0)
+                    {
+                        // Update LifeState on next frame to make sure everything listening on it has time to initialize
+                        StartCoroutine(WaitToUpdateLifeState(LifeState.Fainted));
+                    }
```

Comment:
Couldn't this coroutine be removed by applying the same pattern to `ServerAnimationHandler`? It seems to register itself to the `OnValueChanged` callback for `LifeState`, but doesn't initialize itself inside `OnNetworkSpawn()`.

Reply:
Sorry, what I meant is that I tried to directly call `OnValueChanged` inside `ServerAnimationHandler`'s `OnNetworkSpawn()`. That is what led to the bug where the animation would only play on clients, and not on the host. From what I understand, it seems to be because the `NetworkAnimator`'s `OnNetworkSpawn()` is called after `ServerAnimationHandler`'s `OnNetworkSpawn()`.

Reply:
If that's the case, what do you think about moving the coroutine to that script instead? Let this class set the `LifeState` here and isolate that as a `NetworkAnimator` issue. We can create a Jira to look into that order of execution.

Reply:
Makes sense!

Reply:
For visibility: [ticket](https://jira.unity3d.com/browse/MTT-3152).